### PR TITLE
Fix memory consumption when `max_block_size` is huge in Aggregator + more memory checks

### LINKED ISSUE #48173: 02151_hash_table_sizes_stats triggers OOM
Examples:
https://s3.amazonaws.com/clickhouse-test-reports/0/5fa519d04311615433ad7e016d974de6f7af9860/stateless_tests__tsan__[3/3].html
https://s3.amazonaws.com/clickhouse-test-reports/0/278b8f74c2a54e832cb25c9cd629f6b1939dcfb6/stateless_tests__tsan__[3/3].html

See also https://github.com/ClickHouse/ClickHouse/pull/45181#issuecomment-1380411041

Seems like `02151_hash_table_sizes_stats_distributed` triggers OOM as well. But it was disabled in https://github.com/ClickHouse/ClickHouse/pull/45287, and then we forgot about it. We should try to enable it back.

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed memory consumption in `Aggregator` when `max_block_size` is huge.

Fixes: #48173

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit 5d4347aeb7518a09e575137e92f223c5ecb7662d with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51566/5d4347aeb7518a09e575137e92f223c5ecb7662d/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by nickitat** [comment]
aggregation_by_partitions_11_CPU_select a from t_10000000_16 group by a format Null settings optimize_aggregation_in_ord diff

**Review by alexey-milovidov — APPROVED** [review]

File: tests/queries/0_stateless/02797_aggregator_huge_mem_usage_bug.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+DROP TABLE IF EXISTS v;
+
+create view v (s LowCardinality(String), n UInt8) as select 'test' as s, toUInt8(number) as n from numbers(10000000);
+
+-- this is what allows mem usage to go really high
+set max_block_size=10000000000;
+
+set max_memory_usage_for_user = '1Gi';
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/02797_aggregator_huge_mem_usage_bug.sql:8]: Why "for user"?
It prevents running the test in parallel.

**Comment by alexey-milovidov** [comment]
@nickitat one integration test looks suspicious.

**Comment by nickitat** [comment]
> @nickitat one integration test looks suspicious.

this test is flaky

**Pull request merged**